File: python/gpt/p4_nre.py
def main() -> None:
    vehicles = ["Car", "Bike", "Truck", "Bus"]

    try:
        raw = input("Enter vehicle index: ")
    except EOFError:
        print("No input provided")
        return

    try:
        idx = int(raw.strip())
    except ValueError:
        print("Invalid index")
        return

    if idx < 0 or idx >= len(vehicles):
        print("Index out of range")
        return

    print(f"Vehicle: {vehicles[idx]}")


if __name__ == "__main__":
    main()

File: java/gpt/p4_nre.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class p4_nre {
    public static void main(String[] args) throws IOException {
        String[] vehicles = {"Car", "Bike", "Truck", "Bus"};

        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        System.out.print("Enter vehicle index: ");
        String line = reader.readLine();
        if (line == null) {
            System.err.println("No input provided");
            return;
        }

        int idx;
        try {
            idx = Integer.parseInt(line.trim());
        } catch (NumberFormatException ex) {
            System.err.println("Invalid index");
            return;
        }

        if (idx < 0 || idx >= vehicles.length) {
            System.err.println("Index out of range");
            return;
        }

        System.out.println("Vehicle: " + vehicles[idx]);
    }
}
